### ISSUE TITLE
[pro] Arbitrary Order breaks stable cursors

### ISSUE THREAD

**Issue by wakeless** [rmosolgo/graphql-ruby#2152]
I'm trying to order by a derived column:

```
order("COALESCE(timestamp_a, current_timestamp) - timestamp_b) DESC")
```

This breaks when generating the stable cursors as the rudimentary SQL parser does not understand.

It would be great is we could use the magic of `Arel::Nodes::Ascending` and `Arel::Nodes::SqlLiteral` to allow this to happen. The following code in relation_connection/order:

```
    when Arel::Nodes::Ascending
      if v.expr.is_a? Arel::Nodes::SqlLiteral
        normalized_orders << NormalizedOrder.new(
          v.expr,
          false,
          :asc
        )
      else
        normalized_orders << NormalizedOrder.new(
          v.expr.name.to_s,
          v.expr.relation.name,
          :asc,
        )

      end
    when Arel::Nodes::Descending
      if v.expr.is_a? Arel::Nodes::SqlLiteral
        normalized_orders << NormalizedOrder.new(
          v.expr,
          false,
          :desc
        )
      else
        normalized_orders << NormalizedOrder.new(
          v.expr.name.to_s,
          v.expr.relation.name,
          :desc,
        )
        end
```

Allows me to change my code to:
```
order(Arel.sql("COALESCE(timestamp_a, current_timestamp) - timestamp_b)").desc)
```

And work without problem

**Comment by wakeless** [comment]
@rmosolgo Sorry to bug you, but what's the ETA on this? Do you need me to help with anything further?

**Comment by rmosolgo** [comment]
Hey! Thanks for the ping here. I misread the issue originally and thought that you had a workaround for now. 

I just released 1.10.3 with a patch like the one you suggested (thanks for that!), want to give it a try and follow up here if you find any trouble with it?
